src/System.Slices/System/Span.cs

```diff
@@ -400,6 +400,39 @@ public override bool Equals(object obj)
             return false;
         }
 
+        /// <summary>
+        /// Determines whether the current span is a slice of the supplied span
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public unsafe bool IsSliceOf(Span<T> parentSpan)
+        {
+            var elementSize = PtrUtils.SizeOf<T>();
+            ulong parentStart = parentSpan.Offset.ToUInt64(), thisStart = Offset.ToUInt64();
+            return Object == parentSpan.Object // either both pointers (both null), or same array instance
+                && parentStart <= thisStart // parent must start earlier (or equal start)
+                && (parentStart + ((ulong)parentSpan.Length * (ulong)elementSize)) // parent must  end later (or equal end)
+                    >= (thisStart + ((ulong)Length * (ulong)elementSize))
+                && (Object != null || (((thisStart - parentStart) % (ulong)elementSize) == 0)); // pointer spans must have equal alignment
+        }
+
+        /// <summary>
+        /// Determines whether the current span is a slice of the supplied span
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public bool IsSliceOf(Span<T> parentSpan, out int start)
```

Comment:
`IsSliceOf` does suggest a parent child relationship; which may not be how they came to be.

Invert the calls?

`public bool Contains(Span<T> span, out int start)`